sql/parser/builtins.go

```diff
@@ -943,6 +961,125 @@ func datumToString(datum Datum) (string, error) {
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
 
+var replaceSubRe = regexp.MustCompile(`\\[&1-9]`)
+
+func regexpReplace(s, pattern, to, sqlFlags string) (Datum, error) {
+	pattern, global, err := regexpEvalFlags(pattern, sqlFlags)
+	if err != nil {
+		return nil, err
+	}
+
+	patternRe, err := regexp.Compile(pattern)
```

Comment:
To check against `regexp.String` you'll have to convert the pattern to a regex again which we'd want to avoid.